scripts/chronicle_dump/chronicle_dump.erl

```diff
@@ -150,6 +150,51 @@ sanitize_opt(Value) ->
             {error, invalid_function}
     end.
 
+binarify_output_item(Binary) when is_binary(Binary) -> Binary;
+binarify_output_item(Atom) when is_atom(Atom) -> atom_to_binary(Atom, latin1);
+binarify_output_item(Int) when is_integer(Int) -> integer_to_list(Int);
+binarify_output_item(String) when is_list(String) -> String.
+
+dump_guts(Args) ->
+    {Path, _Options} = parse_args(Args, #{}),
+    Guts = dump_guts_inner(Path),
+    Items = [E || {K, V} <- Guts, E <- [K, V]],
+    ?fmt("~s", [[[binarify_output_item(Item) | <<0:8>>] || Item <- Items]]).
+
+dump_guts_inner(Path) ->
+    case chronicle_storage:read_rsm_snapshot(Path) of
+        {ok, Snapshot} ->
+            try
+                Props0 = chronicle_rsm:format_snapshot(Snapshot),
+                Props1 = proplists:get_value("RSM state", Props0),
+                {?RAW_TAG, Props} = Props1,
+
+                CompatVer = get_value(cluster_compat_version, Props),
+                BucketNames = get_value(bucket_names, Props),
+
+                [{cluster_compat_version,
+                  iolist_to_binary(io_lib:format("~p", [CompatVer]))},
+                 {bucket_names, string:join(BucketNames, ",")},
+                 {rebalance_type, get_value(rebalance_type, Props)}]
+            catch
+                T:E:Stacktrace ->
+                    ?error("Unexpected exception: ~p:~p. Stacktrace:~n"
+                           "~p",
+                           [T, E,
+                            chronicle_utils:sanitize_stacktrace(Stacktrace)])
+            end;
+        {error, Error} ->
+            ?error("Couldn't read snapshot '~s': ~w", [Path, Error])
+    end.
+
+get_value(Key, Props) ->
+    case proplists:get_value(Key, Props) of
+        {Value, _ChronicleMeta} ->
+            Value;
+        _ ->
```

Comment:
can this be anything other than undefined?

Reply:
Artem responded via slack
```
I prefer this if we know that it has to be undefined or tuple. Erlang code should crash if unexpected happens
```